ignite/metrics/metric.py

```diff
@@ -108,14 +118,31 @@ def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union
         else:
             raise TypeError("Unhandled input type {}".format(type(tensor)))
 
-        # synchronize and reduce
-        dist.barrier()
-        dist.all_reduce(tensor)
+        reduction(tensor)
 
         if tensor_to_number:
             return tensor.item()
         return tensor
 
+    @staticmethod
+    def _tpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        xm.all_reduce("sum", [tensor,])
+
+    @staticmethod
+    def _gpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        dist.barrier()
```

Comment:
I think this barrier is useless. The following all_reduce() is a sync collective call.

Reply:
Since it's there currently, I'd be a bit wary of changing the behavior in this PR. Since a lot of things are changing, I'd hate to have less information about potential breakages.

Reply:
I agree!! 😊